Mobile view styling

micahmills opened the pull request:
This contains several minor changes that @mikeallbutt found during mobile testing.

It also contains a change to the top and second bar. First, I fixed the sticky headers so they work properly on all sizes. To do I used the "mobile" top blue bar on medium size screens because it was causing weird wrapping on the smallest medium screens.

I also added the record title into the second bar to make it easier to see what record you are working on if you are scrolled down the page. 

In adding the title I realized that I could also add all the data that was in the contact-type tile and animate the unneeded information away when the users scrolls and the menu is stuck to the top. Because I moved the contact-type tile info to the second bar, I deleted the contact-type tile. 

@corsacca you might have a different opinion on this, but I like having the data in the top bar instead of another tile. That being said, we could also keep the tile and only display it on medium to large screens and save the space on mobile. Or we could just put the record name in the bar and leave the tile in place.

corsacca commented:
Before Mobile:
![image](https://user-images.githubusercontent.com/24901539/105690829-c08a7100-5efc-11eb-8df9-e2301f672520.png)
After Mobile:
![image](https://user-images.githubusercontent.com/24901539/105691006-fd566800-5efc-11eb-9c41-0e462695ece9.png)
![image](https://user-images.githubusercontent.com/24901539/105691038-06dfd000-5efd-11eb-9166-145ad1490e7f.png)

corsacca commented:
Before Desktop:
![image](https://user-images.githubusercontent.com/24901539/105691301-5b834b00-5efd-11eb-9a03-585104e0cef5.png)


After Desktop:
![image](https://user-images.githubusercontent.com/24901539/105691256-4c040200-5efd-11eb-8755-6c32a1ee0af4.png)

corsacca commented:
If you refresh on mobile when while already scrolled down, and then scroll back up:
![image](https://user-images.githubusercontent.com/24901539/105691534-a7ce8b00-5efd-11eb-992e-706ea4c4c855.png)
The top blue bar is missing.

corsacca commented:
The name tile and person icon change color depending on the contact type. I'd like to keep something similar:
![image](https://user-images.githubusercontent.com/24901539/105691858-098ef500-5efe-11eb-8043-64ef1be0c46f.png)
![image](https://user-images.githubusercontent.com/24901539/105691899-16134d80-5efe-11eb-913a-c17745d1b776.png)
![image](https://user-images.githubusercontent.com/24901539/105691911-1dd2f200-5efe-11eb-9367-a5d86f628a85.png)

After stays the same:
![image](https://user-images.githubusercontent.com/24901539/105692056-40fda180-5efe-11eb-9f4d-ab2cf536b992.png)

corsacca commented:
@micahmills Great questions and suggestions.
I like where you are going with this. 
I really like how the sticky bar on mobile and the extra fields disappearing.
I've added screen shots above to visualize what is happening.
We could also play with putting the "[Type] record, created on ..." in the details tile.

corsacca commented:
![image](https://user-images.githubusercontent.com/24901539/105692637-efa1e200-5efe-11eb-96dd-0536a632e49e.png)
Maybe beyond the scope of this pr: I we might need to put "share, following, and tasks" in a dropdown menu to clean up the ui a bit on mobile.

corsacca commented:
Note, in the tile the record name is editable.

micahmills commented:
I have added the icon color as well as added the icon to the mobile view. I also reimplemented the title editable from the secondary bar. I also fixed the bug that caused the menu not to appear if it loaded while scrolled down the page.

Something to not, the sticky bars can possible get messed up  if you resize the window but usually a scroll to the top or refresh fixes that. I don't really consider that a bug, because most users don't resize windows on a regular basis.

micahmills commented:
I like the idea of consolidating the action buttons on the mobile view, but I also like the comments button that makes it quick to scroll to the comment section. I think we should probably make a separate issue/PR for that and see if we can work something out there.

corsacca commented on single-template.php:
```diff
@@ -46,45 +46,19 @@
-                            </h4>
-                            <p><?php esc_html_e( 'Please provide an update by posting a comment.', 'disciple_tools' )?></p>
-                        </div>
+                        <a href="#comment-activity-section">
```
@micahmills Brilliant!
this one is a bit weird on desktop, is just scrolls the page down some:

corsacca replied:
![image](https://user-images.githubusercontent.com/24901539/106268496-7fab9880-622b-11eb-84af-1e185b053550.png)

micahmills replied:
Yeah that would be a bit weird because presumably the comments section is on the screen in the second column on the desktop view. Let me think about this and I am sure we can come up with a way to handle it.

micahmills commented on single-template.php:
```diff
@@ -55,6 +55,13 @@
+                                    <?php echo esc_html( sprintf( __( 'This %s needs an update.', 'disciple_tools' ), strtolower( $post_settings["label_singular"] ) ) ) ?>
+                                </h4>
+                                <p><?php esc_html_e( 'Please provide an update by posting a comment.', 'disciple_tools' )?></p>
+                            </div>
```
Not the most elegant solution, but it works. If we can think of a way to do this without having to duplicate the notification box HTML that would be ideal, but we are doing the same things with menus and other places so it isn't the end of the world.

corsacca replied:
Thanks! This works great

corsacca commented:
Noticed it here, but not a result of this PR. The Person icon is also used on  groups and  trainings. Before we had different icons.
